orchestrator/batch_request.go

```diff
@@ -0,0 +1,143 @@
+	)
+}
+
+type batchRequester struct {
+	log                  log.Logger
+	retries              uint
+	minBatchFee          float64
+	erc20ContractMapping map[eth.Address]string
+}
+
+func (r *batchRequester) run(
+	ctx context.Context,
+	injective InjectiveNetwork,
+	feed PriceFeed,
+) error {
+	r.log.WithField("min_batch_fee", r.minBatchFee).Infoln("scanning Injective for potential batches")
+
+	unbatchedFees, err := r.getUnbatchedFeesByToken(ctx, injective)
+	if err != nil {
+		// non-fatal, just alert
+		r.log.WithError(err).Warningln("unable to get unbatched fees from Injective")
+		return nil
+	}
+
+	if len(unbatchedFees) == 0 {
+		r.log.Debugln("no outgoing withdrawals or minimum batch fee is not met")
+		return nil
+	}
+
+	for _, tokenFee := range unbatchedFees {
+		r.requestBatchCreation(ctx, injective, feed, tokenFee)
+	}
+
+	return nil
+}
+
+func (r *batchRequester) getUnbatchedFeesByToken(ctx context.Context, injective InjectiveNetwork) ([]*types.BatchFees, error) {
+	var unbatchedFees []*types.BatchFees
+	retryFn := func() (err error) {
+		unbatchedFees, err = injective.UnbatchedTokenFees(ctx)
+		return err
+	}
+
+	if err := retry.Do(retryFn,
+		retry.Context(ctx),
+		retry.Attempts(r.retries),
+		retry.OnRetry(func(n uint, err error) {
+			log.WithError(err).Errorf("failed to get unbatched fees, will retry (%d)", n)
+		}),
+	); err != nil {
+		return nil, err
+	}
+
+	return unbatchedFees, nil
+}
+
+func (r *batchRequester) requestBatchCreation(
+	ctx context.Context,
+	injective InjectiveNetwork,
+	feed PriceFeed,
+	batchFee *types.BatchFees,
+) {
+	var (
+		tokenAddr = eth.HexToAddress(batchFee.Token)
+		denom     = r.tokenDenom(tokenAddr)
+	)
+
+	if thresholdMet := r.checkFeeThreshold(feed, tokenAddr, batchFee.TotalFees); !thresholdMet {
+		r.log.WithFields(log.Fields{
+			"denom":          denom,
+			"token_contract": tokenAddr.String(),
+			"total_fees":     batchFee.TotalFees.String(),
+		}).Debugln("skipping underpriced batch")
+		return
+	}
+
+	r.log.WithFields(log.Fields{
+		"denom":          denom,
+		"token_contract": tokenAddr.String(),
+	}).Infoln("requesting batch creation on Injective")
+
+	_ = injective.SendRequestBatch(ctx, denom)
+}
+
+func (r *batchRequester) tokenDenom(tokenAddr eth.Address) string {
+	if cosmosDenom, ok := r.erc20ContractMapping[tokenAddr]; ok {
+		return cosmosDenom
+	}
+
+	// peggy denom
+	return types.PeggyDenomString(tokenAddr)
+}
+
+func (r *batchRequester) checkFeeThreshold(
+	feed PriceFeed,
+	tokenAddr eth.Address,
+	totalFees cosmtypes.Int,
+) bool {
+	if r.minBatchFee == 0 {
+		return true
+	}
+
+	tokenPriceInUSD, err := feed.QueryUSDPrice(tokenAddr)
+	if err != nil {
+		return false
+	}
+
+	tokenPriceInUSDDec := decimal.NewFromFloat(tokenPriceInUSD)
+	totalFeeInUSDDec := decimal.NewFromBigInt(totalFees.BigInt(), -18).Mul(tokenPriceInUSDDec)
+	minFeeInUSDDec := decimal.NewFromFloat(r.minBatchFee)
+
+	if totalFeeInUSDDec.GreaterThan(minFeeInUSDDec) {
```

Comment:
Are you sure this works? The scaling doesn't seem to be applied, e.g. using USDT or even INJ as an example. 

FYI I'm also aware that `decimal.NewFromBigInt(totalFees.BigInt(), -18)` is necessary when converting the sdk.Int to decimal (can be extracted to a helper function for clarity) @dbrajovic

Reply:
If it doesn't work then it never did - this is the original implementation. 

I can write up some unit tests to see if scaling works for some given input.